### PR TITLE
Cast and Crew Buttons

### PR DESCRIPTION
**Pull request by TheDoplarEffect**
Cast and Crew in the Cast.tsx routes folder.

File uses two sub components, the cast and Crew Buttons and the Movies Action Button which is just them with a square.

When running on vite server works (http://127.0.0.1:5174/movie/500/castcrew)

Also added some animations from switching between pages.

On figma file the scrollable list didn't reach to the end of the page so whole component had to be wrapped in a div with a bottom padding.  However, using the measurements from the figma file it doesn't look anywhere close.  Used pb-12 instead since it looked the closest to original design.

### CODE REVIEW

**Comment by TheDoplarEffect** [comment]
Things I forgot to mention:

- Added custom classes to css to remove scrollbar
- added placeholder image incase profile picture can't be loaded (common with smaller movies or crew members)
- Added a new function to the API

File: src/components/CastCrew.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import clsx from "clsx";
+import { useState } from "react";
+
+type input = {
```
Comment (dnmct) [inline src/components/CastCrew.tsx:4]: Pls titlecase Types
Convention: Append Props to the typename to make it clear

File: src/components/CastCrew.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  input: string;
+  setInput: React.Dispatch<React.SetStateAction<string>>;
+};
+export function CastCrewButtons({ input, setInput }: input) {
+  const active = "text-white border-white bg-white-dimmed";
+  const inactive = "text-white-dimmed bg-dark-light border-dark-light";
+  return (
+    <div className="flex justify-between sticky top-0 bg-dark z-10">
+      <p
```
Comment (dnmct) [inline src/components/CastCrew.tsx:13]: Make that a button if it's intended to be clickable

File: src/components/MoviesActionButton.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+type Input = {
```
Comment (dnmct) [inline src/components/MoviesActionButton.tsx:1]: the name MoviesActionButton is a bit misleading since this component is not interactive

**Review by dnmct — CHANGES_REQUESTED** [review]

File: src/components/CreditsProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+type Input = {
+  image: string;
+  name: string;
+  role: string;
+};
+export default function MoviesActionButton({ image, name, role }: Input) {
+  return (
+    <div className="flex items-center gap-[1.25rem]">
```
Comment (ReallyBoard123) [inline src/components/CreditsProfile.tsx:8]: Isn`t there a tailwind class here that can be used? or was it in order to keep it consistent to the figma design?

**Review by ReallyBoard123 — CHANGES_REQUESTED** [review]
Everything looks great! It works as intended. I love the creative touch of the animation. There is a slight inconsistency when one switches from Cast to Crew and back to Cast; the animation ends abruptly or relatively quickly.

Just see if there is an alternative to the tailwind class for the [1.25 rem] you used there. Otherwise, all good.

**Comment by TheDoplarEffect** [comment]
Changed the custom file.
The jittering in the animation is caused when someone clicks on the enxt button before the animation has finished.

**Review by ReallyBoard123 — APPROVED** [review]

**Review by dnmct — APPROVED** [review]

**Pull request merged**